modules/virt-accessing-rdp-console.adoc

```diff
@@ -6,71 +6,101 @@
 
 .Prerequisites
 
-* A running Windows virtual machine with the QEMU guest agent installed. The
-`qemu-guest-agent` is included in the VirtIO drivers.
-* A layer 2 NIC attached to the virtual machine. 
-* An RDP client installed on a machine on the same network as the
-Windows virtual machine.
+* A running Windows virtual machine with the QEMU guest agent installed. The `qemu-guest-agent` is included in the VirtIO drivers.
+* An RDP client installed on a machine on the same network as the Windows VM.
 
 .Procedure
 
-. Log in to the {VirtProductName} cluster through the `oc` CLI tool as a user with
-an access token.
+. Edit the `VirtualMachine` manifest to add the label for service creation:
++
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: vm-ephemeral
+  namespace: example-namespace
+spec:
+  running: false
+  template:
+    metadata:
+      labels:
+        special: key <1>
+# ...
+----
+<1> Add the label `special: key` in the `spec.template.metadata.labels` section.
++
+
+[NOTE]
+====
+Labels on a virtual machine are passed through to the pod. The `special: key` label must match the label in the `spec.selector` attribute of the `Service` manifest.
+====
+
+. Save the `VirtualMachine` manifest file to apply your changes.
+. Create a `Service` manifest to expose the VM:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: vmservice <1>
+  namespace: example-namespace <2>
+spec:
+  ports:
+  - targetPort: 3389 <3>
+    protocol: TCP
+    port: 27017
+  selector:
+    special: key <4>
+  type: NodePort <5>
+# ...
+----
+<1> The name of the `Service` object.
+<2> The namespace where the `Service` object resides. This must match the `metadata.namespace` field of the `VirtualMachine` manifest.
+<3> The VM port to be exposed by the service. It must reference an open port if a port list is defined in the VM manifest. If `targetPort` is not specified, it takes the same value as `port`.
+<4> The reference to the label that you added in the `spec.template.metadata.labels` stanza of the `VirtualMachine` manifest.
+<5> The type of service.
+
+. Save the `Service` manifest file.
+. Create the service by running the following command:
 +
 [source,terminal]
 ----
-$ oc login -u <user> https://<cluster.example.com>:8443
+$ oc create -f <service_name>.yaml
 ----
 
-. Use `oc describe vmi` to display the configuration of the running
-Windows virtual machine.
+. Start the VM. If the VM is already running, restart it.
+. Query the `Service` object to verify that it is available:
 +
+[source, terminal]
+----
+$ oc get service -n example-namespace
+----
++
+.Example output for `NodePort` service
 [source,terminal]
 ----
-$ oc describe vmi <windows-vmi-name>
+NAME        TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)            AGE
+vmservice   NodePort    172.30.232.73   <none>       27017:30000/TCP    5m
```

Comment:
```suggestion
rdpservice   NodePort    172.30.232.73   <none>       3389:30000/TCP    5m
```

User should write down the randomly assigned port (30000 in this case). It will be later needed together with the node IP.